jrnl/cli.py

```diff
@@ -478,5 +478,8 @@ def run(manual_args=None):
         journal.write()
 
     elif args.delete:
+        other_entries = [e for e in old_entries if e not in journal.entries]
         journal.prompt_delete_entries()
+        journal.entries += other_entries
+        journal.sort()
```

Comment:
All `sort`s should happen in the open method for a journal. By keeping them there, we avoid resorting (again, this helps very large journals).